backend/getDisplayName.js

```diff
@@ -18,7 +18,23 @@ function getDisplayName(type: Function): string {
     return cachedDisplayNames.get(type);
   }
 
-  let displayName = type.displayName || type.name || 'Unknown';
+  let displayName = type.displayName || type.name;
+
+  // The displayName property might not be a string.
+  // If it's a custom function, attempt to use its return value.
+  // github.com/facebook/react-devtools/issues/803
+  if (typeof displayName === 'function') {
+    try {
+      displayName = displayName();
```

Comment:
Sure, that's fair. I can change this to fall back to `name || 'Unknown'` if `displayName` is a function rather than trying to be too clever about it.